python/pyarrow/table.pxi

```diff
@@ -183,38 +183,21 @@ cdef class ChunkedArray(_PandasConvertible):
         return result
 
     def _to_pandas(self, options, **kwargs):
-        cdef:
-            PyObject* out
-            PandasOptions c_options = _convert_pandas_options(options)
-            ChunkedArray array
-
-        if self.type.id == _Type_TIMESTAMP and self.type.unit != 'ns':
-            # pandas only stores ns data - casting here is faster
-            array = self.cast(timestamp('ns'))
-        else:
-            array = self
-
-        with nogil:
-            check_status(libarrow.ConvertChunkedArrayToPandas(
-                c_options,
-                array.sp_chunked_array,
-                array, &out))
-
-        result = pandas_api.series(wrap_array_output(out), name=self._name)
-
-        if isinstance(self.type, TimestampType) and self.type.tz is not None:
-            from pyarrow.pandas_compat import make_tz_aware
-
-            result = make_tz_aware(result, self.type.tz)
-
-        return result
+        return _array_like_to_pandas(self, options)
 
     def __array__(self, dtype=None):
         cdef:
             PyObject* out
             PandasOptions c_options
             object values
 
+        if self.type.id == _Type_EXTENSION:
+            return (
+                chunked_array(
+                    [self.chunk(i).storage for i in range(self.num_chunks)]
+                ).__array__(dtype)
```

Comment:
Was this already tested?

Reply:
Seems it is tested here https://github.com/apache/arrow/blob/master/python/pyarrow/tests/test_extension_type.py#L394